### PR TITLE
Allow options to disable user ssh keys configuration from the interface on app.ini

### LINKED ISSUE #24716: Allow disabling users from adding SSH keys
### Feature Description

It would be great if we have the option to disable SSH key adding by users, for when keys are entirely externally managed by an admin. Seems like removing the front end button is easy enough using a template, but from my understanding, to be secure we will also need to disable the API endpoint. I don't think such a feature exists right now. 

In my mind, an implementation could be to

1. Add an SSH configuration option to "allow non-admin users to add SSH keys" (default to true).
2. Make the front end check for this setting and set the "Add SSH Key" button to disabled. 
3. Make the api endpoint check this setting.

Would this be a good idea? Or is there a configuration to achieve this already? If this will be a good contribution, I plan to make a PR of it.

Issue #13983 seems relevant, but it only deals with deleting keys.

(I have [asked this on Discord](https://discord.com/channels/322538954119184384/322910365237248000/1107263750723076216) and was advised to make a proposal here)

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by lunny**
Follow #29275
Extract from #20549
Fix #24716

### CODE REVIEW

File: custom/conf/app.example.ini [hunk 1]
```diff
@@ -1474,8 +1474,9 @@ LEVEL = Info
 ;;
 ;; Default configuration for email notifications for users (user configurable). Options: enabled, onmention, disabled
 ;DEFAULT_EMAIL_NOTIFICATIONS = enabled
-;; Disabled features for users, could be "deletion", more features can be disabled in future
+;; Disabled features for users, could be "deletion", "ssh_keys" more features can be disabled in future
 ;; - deletion: a user cannot delete their own account
+;; - ssh_keys: a user cannot configuration ssh keys
```
Comment (delvh) [inline custom/conf/app.example.ini]:
```suggestion
;; - ssh_keys: a user cannot configure ssh keys
```

File: docs/content/administration/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -518,8 +518,9 @@ And the following unique queues:
 
 - `DEFAULT_EMAIL_NOTIFICATIONS`: **enabled**: Default configuration for email notifications for users (user configurable). Options: enabled, onmention, disabled
 - `DISABLE_REGULAR_ORG_CREATION`: **false**: Disallow regular (non-admin) users from creating organizations.
-- `USER_DISABLED_FEATURES`: **_empty_** Disabled features for users, could be `deletion` and more features can be added in future.
+- `USER_DISABLED_FEATURES`: **_empty_** Disabled features for users, could be `deletion`, `ssh_keys` and more features can be added in future.
   - `deletion`: User cannot delete their own account.
+  - `ssh_keys`: User cannot configuration ssh keys.
```
Comment (delvh) [inline docs/content/administration/config-cheat-sheet.en-us.md]:
```suggestion
  - `ssh_keys`: User cannot configure ssh keys.
```

File: docs/content/administration/config-cheat-sheet.zh-cn.md [hunk 3]
```diff
@@ -497,8 +497,9 @@ Gitea 创建以下非唯一队列：
 
 - `DEFAULT_EMAIL_NOTIFICATIONS`: **enabled**：用户电子邮件通知的默认配置（用户可配置）。选项：enabled、onmention、disabled
 - `DISABLE_REGULAR_ORG_CREATION`: **false**：禁止普通（非管理员）用户创建组织。
-- `USER_DISABLED_FEATURES`:**_empty_** 禁用的用户特性，当前允许为空或者 `deletion`， 未来可以增加更多设置。
+- `USER_DISABLED_FEATURES`:**_empty_** 禁用的用户特性，当前允许为空或者 `deletion`，`ssh_keys` 未来可以增加更多设置。
   - `deletion`: 用户不能通过界面或者API删除他自己。
+  - `ssh_keys`: 用户不能通过界面配置SSH Keys。
```
Comment (a1012112796) [inline docs/content/administration/config-cheat-sheet.zh-cn.md]: API?

Comment (lunny) [reply docs/content/administration/config-cheat-sheet.zh-cn.md]: ~I think there is no API to allow users to update ssh keys currently.~

Comment (lunny) [reply docs/content/administration/config-cheat-sheet.zh-cn.md]: Done.

File: routers/api/v1/user/key.go [hunk 4]
```diff
@@ -198,6 +199,11 @@ func GetPublicKey(ctx *context.APIContext) {
 
 // CreateUserPublicKey creates new public key to given user by ID.
 func CreateUserPublicKey(ctx *context.APIContext, form api.CreateKeyOption, uid int64) {
+	if setting.Admin.UserDisabledFeatures.Contains(setting.UserFeatureSSHKeys) {
```
Comment (delvh) [inline routers/api/v1/user/key.go]: Sounds like SSH would be disabled completely.
I wonder, should we rename the keys to make it more obvious that only the configuration is forbidden?

File: routers/api/v1/user/key.go [hunk 5]
```diff
@@ -198,6 +199,11 @@ func GetPublicKey(ctx *context.APIContext) {
 
 // CreateUserPublicKey creates new public key to given user by ID.
 func CreateUserPublicKey(ctx *context.APIContext, form api.CreateKeyOption, uid int64) {
```
Comment (delvh) [inline routers/api/v1/user/key.go:201]: [DELETE `/user/keys/{id}`](https://gitea.com/api/swagger#/user/userCurrentDeleteKey) is missing

File: routers/api/v1/user/key.go (hunk 4 above)
Comment (jolheiser) [reply routers/api/v1/user/key.go]: I think in context it probably makes enough sense, and presumably an admin would have to read the docs to find the key in the first place. 
But, an alternative name could be `manage_ssh_keys` or similar?

Comment (lunny) [reply routers/api/v1/user/key.go]: Done.

File: routers/api/v1/user/key.go (hunk 5 above)
Comment (lunny) [reply routers/api/v1/user/key.go:201]: Done.

**Review by delvh — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by jackHay22** [comment]
@lunny I opened a PR to add an additional setting that disables user features for external logins: https://github.com/go-gitea/gitea/pull/29615

**Pull request merged**